src/Extension/CoreExtension.php

```diff
@@ -2055,4 +2073,27 @@ public static function parseAttributeFunction(Parser $parser, Node $fakeNode, $a
 
         return new GetAttrExpression($args[0], $args[1], $args[2] ?? null, Template::ANY_CALL, $line);
     }
+
+    private static function getPropertyChecker(string $class, string $property): \Closure
+    {
+        static $classReflectors = [];
+
+        $class = $classReflectors[$class] ??= new \ReflectionClass($class);
+
+        if (!$class->hasProperty($property)) {
+            static $propertyExists;
+
+            return $propertyExists ??= \Closure::fromCallable('property_exists');
```

Comment:
Do we need a local cache of the closure ? isn't PHP optimizing this internally already ?

Reply:
nope, php is creating a new object each time